Formula/cluster-api.rb

```diff
@@ -0,0 +1,18 @@
+class ClusterApi < Formula
+  desc "Home for the Cluster Management API work, a subproject of sig-cluster-lifecycle"
+  homepage "https://cluster-api.sigs.k8s.io"
+  url "https://github.com/kubernetes-sigs/cluster-api/archive/refs/tags/v0.3.14.tar.gz"
+  sha256 "fc7c7eb07bce70505721c8bc82ff4b8f111172cdfef94964b7ceac82e45faace"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", bin/"clusterctl"
+  end
+
+  test do
+    help_output = shell_output("#{bin}/clusterctl --help 2>&1", 2)
+    assert_match "Usage of #{bin}/clusterctl", help_output
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Yes sure! Thanks!! 🙏

Reply:
Any suggestion on this? This tool is similar to [kubernetes-cli](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb) and, in this formular, their are checking for help and version.

I can check for an error, but this cli needs a kubernetes cluster to work.

Reply:
We have a lot of formulae (usually old ones) with bad tests. We try to fix the ones we can, but there are a lot of them.

Checking for an expected error from running without a kubernetes cluster is fine.

Reply:
Ok thanks for help! I'll work on it!

Reply:
What do you think?